clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -653,6 +663,10 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                       << "marker type " << memref.marker.marker_type << " value 0x"
                       << std::hex << memref.marker.marker_value << std::dec << "\n";
         }
+        // Push the return address when kernel transfer happens.
+        if (memref.marker.marker_type == TRACE_MARKER_TYPE_KERNEL_EVENT) {
+            shard->retaddr_stack_.push(memref.marker.marker_value);
```

Comment:
The interruption point is not a target of a return, though.  The comment needs to be updated to explain that this is to match the return used post-signal-handler to run the restorer code.  It should also explain that we thus assume all signal handlers return normally and don't longjmp (we would need to store the stack pointer in the stack and detect skipped frames to handle that).

As for the value pushed, probably better to push some sentinel or 0 or something, as the current value you are pushing here is just going to be confusing.

Reply:
I've updated the comment and the code accordingly.